Blockzilla/BrowserViewController.swift

```diff
@@ -251,31 +250,35 @@ class BrowserViewController: UIViewController {
             var biometricError: NSError?
 
             // Check if user is already in a cleared session, or doesn't have biometrics enabled in settings
-            if self.webViewContainer.isHidden || !Settings.getToggle(SettingsToggle.biometricLogin) {
+            if  !Settings.getToggle(SettingsToggle.biometricLogin) || !AppDelegate.needsAuthenticated || self.webViewContainer.isHidden {
+                AppDelegate.splashView?.animateHidden(true, duration: 0.25)
                 return
             }
+            AppDelegate.needsAuthenticated = false
 
-            self.displaySplashScreen()
+            self.context = LAContext()
```

Comment:
This is inside a closure here. It's inside the handler for the notification observer. Is there a different way I should approach this?

Reply:
it is? sorry, github diff view is confusing me, I'll look again at this. This would demonstrate why self is good! `self` clues me in to fact it is a closure-scope, and closures need be reviewed closely for proper object lifetime and thread usage.